Makefile

```diff
@@ -6,21 +6,28 @@ build:
 	bin/docker-prepare && \
 	docker-compose -f docker-compose.test.yml build
 
-run:
+test_variant_app:
 	cd $(APP_NAME) && \
 	docker-compose -f docker-compose.test.yml run test
 
-base_spec = spec/base/**/*_spec.rb,spec/addons/docker/**/*_spec.rb
-test:
+base_addon_spec = spec/addons/base/**/*_spec.rb
+web_addon_spec = spec/addons/variants/web/**/*_spec.rb
+api_addon_spec = spec/addons/variants/api/**/*_spec.rb
+
+base_spec = spec/base/**/*_spec.rb
+web_spec = spec/variants/web/**/*_spec.rb
+api_spec = spec/variants/api/**/*_spec.rb
+
+test_template:
 	cd $(APP_NAME) && \
 	docker-compose -f docker-compose.test.yml up --detach db redis && \
 	docker-compose -f docker-compose.test.yml run --detach test bin/start.sh && \
 	cd ../.template && \
 	bundle install; \
 	if [ $(VARIANT) = web ]; then \
-	  bundle exec rspec --pattern="$(base_spec),spec/variants/web/**/*_spec.rb"; \
+		bundle exec rspec --pattern="$(base_addon_spec),${base_spec},$(web_addon_spec),$(web_spec)"; \
 	elif [ $(VARIANT) = api ]; then \
-	  bundle exec rspec --pattern="$(base_spec),spec/variants/api/**/*_spec.rb"; \
+		bundle exec rspec --pattern="$(base_addon_spec),${base_spec},$(api_addon_spec),$(api_spec)"; \
```

Comment:
Minor bit but the ordering of the spec files (which does not matter to be honest) is a bit counterintuitive in the sense that I would place base_spec first, then variant spec, then base_addon_spec and finally variant_addon_spec (from base to specific variant):

```
bundle exec rspec --pattern="${base_spec}, ${api_spec}, ${base_addon_spec}, $(api_addon_spec)";

bundle exec rspec --pattern="${base_spec}, ${web_spec}, ${base_addon_spec}, $(web_addon_spec)";
```

Reply:
Fixed in https://github.com/nimblehq/rails-templates/pull/248/commits/c87b51926b9f1d643ccb7473b4970c56dbf26aa3 @olivierobert